src/test/java/org/datadog/jmxfetch/JMXServerSupervisorClient.java

```diff
@@ -0,0 +1,57 @@
+package org.datadog.jmxfetch;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class JMXServerSupervisorClient {
+    private final String host;
+    private final int port;
+
+    public JMXServerSupervisorClient(String host, int port) {
+        this.host = host;
+        this.port = port;
+    }
+
+    public void initializeJMXServer(String rmiHostname) throws IOException {
+        String endpoint = "/init";
+        String jsonPayload = "{\"rmiHostname\": \""  + rmiHostname + "\"}";
+        sendPostRequestWithPayload(endpoint, jsonPayload);
+    }
+
+    private void sendPostRequestWithPayload(String endpoint, String jsonPayload) throws IOException {
```

Comment:
This is duplicated from `JMXServerControlClient.java`. I'll extract this to a base-class